packages/api-catalog-service/src/namespace/resolver.ts

```diff
@@ -0,0 +1,128 @@
+import { Namespace } from './schema';
+import { apiCatalogHelper } from '../shared/helpers';
+import { uniq } from 'lodash';
+
+export const NamespaceResolver = {
+    Query: {
+        listNamespaces ( root: any, args: any, ctx: any ) {
+            return Namespace.find().lean().then( async ( namespaces: NamespaceType[] ) => {
+                if ( namespaces.length ) {
+                    // Fetch user information associated with this records.
+                    let userIds: string[] = [];
+                    namespaces.map( ( namespace: NamespaceType ) => {
+                        namespace.createdBy ?
+                            userIds.push( namespace.createdBy )
+                            : null;
+                        namespace.updatedBy ?
+                            userIds.push( namespace.updatedBy )
+                            : null;
+                    } );
+                    userIds = uniq( userIds );
+                    // Build a single user query for the namespace
+                    const userQuery = await apiCatalogHelper.buildUserQuery( userIds );
+                    const userData: UserType[] = await apiCatalogHelper.fetchUserProfile( userQuery );
+
+                    // Assign the parsed user information with the fields.
+                    return namespaces.map( namespace => {
+                        namespace.createdBy ?
+                            namespace.createdBy = userData.filter( user => user.rhatUUID === namespace.createdBy )[ 0 ].mail
```

Comment:
Use `array.find` instead of filter if you only need to find one. It is more optimal and also less prone to errors.

Reply:
Updated